src/Workspaces/Core/Portable/Workspace/Solution/SourceGeneratedDocumentState.cs

```diff
@@ -189,6 +189,27 @@ public SourceGeneratedDocumentState WithGenerationDateTime(DateTime generationDa
             generationDateTime);
     }
 
+    public SourceGeneratedDocumentState WithSyntaxRoot(SyntaxNode newRoot)
+    {
+        var newTextVersion = GetNewerVersion();
+        var newTreeVersion = GetNewTreeVersionForUpdatedTree(newRoot, newTextVersion, PreservationMode.PreserveValue);
+        var treeAndVersion = new TreeAndVersion(newRoot.SyntaxTree, newTreeVersion);
```

Comment:
nit: consider inlining.

Reply:
primarily to make it very clear what is used as temp data, versus what is being passed along to the SourceGeneratedDocumentState .  Note: this is a nit.  if you prefer this it is fine.